events.js

```diff
@@ -0,0 +1,48 @@
+const operatingTimezones = require('./operatingTimezones');
+const isEnabled = require('./isEnabled').isEnabled;
+
+const UTC_START_HOUR = 9;
+const UTC_STOP_HOUR = 22;
+
+function offsetUTCHour(utcHour, utcTimezoneOffset) {
+  const convertedHour = utcHour + utcTimezoneOffset;
+
+  if (convertedHour > 23) {
+    return convertedHour - 23; // wrapped to next day
+  }
+
+  if (convertedHour < 0) {
+    return convertedHour + 23; // wrapped to previous day
+  }
+
+  return convertedHour;
+}
+
+function stop() {
+  const stopCrons = operatingTimezones.map(timezone => ({
+    rate: `cron(30 ${offsetUTCHour(UTC_STOP_HOUR, timezone)} * * ? *)`,
+    enabled: isEnabled(),
+    input: {
+      currentOperatingTimezone: timezone,
+    },
+  }));
+  return {
+    schedule: stopCrons, // CRONS for stopping
+  };
+}
+
+function start() {
+  const startCrons = operatingTimezones.map(timezone => ({
+    rate: `cron(30 ${offsetUTCHour(UTC_START_HOUR, timezone)} * * ? *)`,
+    enabled: isEnabled(),
+    input: {
+      currentOperatingTimezone: timezone,
+    },
+  }));
+  return {
+    schedule: startCrons, // CRONS for starting
+  };
+}
+
+module.exports.stop = stop;
+module.exports.start = start;
```

Comment:
These functions generate the schedule events that end up being put into the serverless.yml on deployment. It works similar to how the dryRun and isEnabled environment variable javascript files work.